native/android/src/main/java/com/nozbe/watermelondb/DatabaseDriver.kt

```diff
@@ -157,13 +170,57 @@ class DatabaseDriver(context: Context, private val configuration: Configuration)
 
     private fun isCached(id: RecordID): Boolean = cachedRecords.contains(id)
 
-    private fun setUpSchema() {
+    private fun setUpSchema(schema: Schema) {
         log?.info("Setting up schema")
-        database.executeSchema(configuration.schema + Queries.localStorageSchema,
-                configuration.schemaVersion)
+        database.executeStatements(schema.sql + Queries.localStorageSchema)
+        database.userVersion = schema.version
     }
 
-    init {
-        setUp()
+    private fun setUpDatabase(schema: Schema) {
+        log?.info("Setting up database with version ${schema.version}")
+        try {
+            unsafeResetDatabase(schema)
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
doesn't it just add things to the console? In Swift, this is a fatal error (ie app crashes), because there's really nothing you can do to fix the situation...